packages/e2e/cypress/e2e/userAttributes.cy.ts

```diff
@@ -81,3 +81,88 @@ describe('User attributes', () => {
         cy.contains('Christina');
     });
 });
+
+describe.only('User attributes dimension required_attribute', () => {
+    beforeEach(() => {
+        cy.login();
+    });
+
+    it('Create customer_id attribute', () => {
+        // This could fail if the attribute already exists
+        cy.request({
+            url: `${apiUrl}/org/attributes`,
+            headers: { 'Content-type': 'application/json' },
+            method: 'POST',
+            body: {
+                name: 'customer_id',
+                users: [
+                    {
+                        userUuid: 'b264d83a-9000-426a-85ec-3f9c20f368ce',
+                        value: '30',
+                    },
+                ],
+            },
+            failOnStatusCode: false,
+        });
+    });
+    it('Delete is_admin attribute', () => {
+        cy.request(`${apiUrl}/org/attributes`).then((resp) => {
+            expect(resp.status).to.eq(200);
+            const customerIdAttr = resp.body.results.find(
+                (attr) => attr.name === 'is_admin',
+            );
+            if (customerIdAttr)
+                cy.request({
+                    url: `${apiUrl}/org/attributes/${customerIdAttr.uuid}`,
+                    method: 'DELETE',
+                }).then((r) => {
+                    expect(r.status).to.eq(200);
+                });
+        });
+    });
+    it('Should not see last_name dimension', () => {
+        cy.visit(`/projects/${SEED_PROJECT.project_uuid}/tables`);
+
+        cy.findByText('Users').click();
+        cy.findByText('Last name').should('not.exist');
+    });
+
+    it('Create user attribute', () => {
+        cy.visit(`/generalSettings/userAttributes`);
+        cy.findByText('Add new attributes').click();
+
+        cy.get('input[name="name"]').type('is_admin');
+        cy.findByText('Add user').click();
+        cy.findByPlaceholderText('E.g. test@lightdash.com').type('demo');
+        cy.findByText('demo@lightdash.com').click();
+        cy.get('input[name="users.0.value"]').type('true');
+        cy.findByText('Add').click();
+        cy.contains('Success');
+    });
+
+    it('Should see last_name attribute', () => {
+        cy.visit(`/projects/${SEED_PROJECT.project_uuid}/tables`);
+
+        cy.findByText('Users').click();
+        cy.findByText('Last name').click();
+
+        // run query
+        cy.get('button').contains('Run query').click();
+        cy.contains('W.');
```

Comment:
I don't know what this is 😅

Reply:
This means the user with id 30 should be able to see its own last_name: 'W.'

Reply:
thanks!